core/src/main/scala/io/prediction/core/CleanedDataSource.scala

```diff
@@ -103,12 +106,29 @@ trait CleanedDataSource {
       } ++ events.filter(!isSetEvent(_))
   }
 
+  //TODO ensure most recent is preserved
   def removePDuplicates(sc: SparkContext, rdd: RDD[Event]): RDD[Event] = {
```

Comment:
Just to clarify eventTime proper is a matching criteria here for de-duping. If 2 events differ by eventTime, they will be considered distinct. However, if they differ only by event Id and/or creationTime, they will be considered dupes. So in regards to creationTime, currently the latest IS preserved. I will add that to the test then remove this comment.